### PR TITLE
HDDS-7783. Add compile platform info in 'ozone version' command output

### PR DESCRIPTION
**Pull request by ChenSammi**
https://issues.apache.org/jira/browse/HDDS-7783

### CODE REVIEW

File: hadoop-hdds/common/src/main/resources/hdds-version-info.properties [hunk 1]
```diff
@@ -25,4 +25,5 @@ url=${version-info.scm.uri}
 srcChecksum=${version-info.source.md5}
 hadoopProtoc2Version=${proto2.hadooprpc.protobuf.version}
 hadoopProtoc3Version=${proto3.hadooprpc.protobuf.version}
-grpcProtocVersion=${grpc.protobuf-compile.version}
\ No newline at end of file
+grpcProtocVersion=${grpc.protobuf-compile.version}
+platform=${os.detected.classifier}
```
Comment (kaijchen) [inline hadoop-hdds/common/src/main/resources/hdds-version-info.properties:29]: Looks like this value is defined in compile time.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/HddsVersionInfo.java [hunk 2]
```diff
@@ -52,6 +52,8 @@ public static void main(String[] args) {
             " and " + HDDS_VERSION_INFO.getHadoopProtoc3Version());
     System.out.println(
         "From source with checksum " + HDDS_VERSION_INFO.getSrcChecksum());
+    System.out.println(
+        "For platform " + HDDS_VERSION_INFO.getPlatform());
```
Comment (kaijchen) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/HddsVersionInfo.java:56]: Correct me if I'm wrong, I don't think the current Ozone binary package is platform specific.
If it's true, we should give information like "compiled on x86_64", but not "compiled for x86_64".

File: hadoop-hdds/common/src/main/resources/hdds-version-info.properties (hunk 1 above)
Comment (ChenSammi) [reply hadoop-hdds/common/src/main/resources/hdds-version-info.properties:29]: Yes, the value is from the below platform.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/HddsVersionInfo.java (hunk 2 above)
Comment (ChenSammi) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/HddsVersionInfo.java:56]: Right, "compiled on" is an accurate wording.

**Comment by adoroszlai** [comment]
Thanks @ChenSammi for the patch.

> https://issues.apache.org/jira/browse/HDDS-7783

Can you please add a description to the PR or the Jira issue, explaining why the platform info is needed?  The PR has no description, only a link to the task, which has no description either, so it's hard to tell what's the goal.

**Comment by kerneltime** [comment]
@tanvipenumudy can you please take a look as well.

**Review by ayushtkn — COMMENTED** [review]
I gave it a try and things work as expected, so prod code looks good
```
Source code repository https://github.com/apache/hadoop-ozone.git -r b9bc6bc89143248dff83e0493db7462f59b043bd
Compiled by ayushsaxena on 2023-01-25T16:34Z
Compiled with protoc 2.5.0, 3.19.6 and 3.7.1
From source with checksum 53463289377e7f44b4fddacf3d88324
With Apache Ratis: 2.4.2-8b8bdda-SNAPSHOT from 8b8bddab8d8aff8566b7ae84c6190344e49f74c7
Compiled on platform osx-x86_64

Using HDDS 1.4.0-SNAPSHOT
Source code repository https://github.com/apache/hadoop-ozone.git -r b9bc6bc89143248dff83e0493db7462f59b043bd
Compiled by ayushsaxena on 2023-01-25T16:29Z
Compiled with protoc 2.5.0, 3.19.6 and 3.7.1
From source with checksum 96454fe64f58376f48933c19806bdbca
Compiled on platform osx-x86_64
```

I am also curious where it would be useful, like do we have some native changes like which are OS/Arch dependent?

**Comment by ChenSammi** [comment]
> 
> I am also curious where it would be useful, like do we have some native changes like which are OS/Arch dependent?

Hi @ayushtkn,  it is just a piece of trivial and interesting info about the compiled platform, not related with any arch dependency.

**Review by ayushtkn — APPROVED** [review]
Thanx @ChenSammi,makes sense to me, it isn't harmless :-) 
+1

**Comment by adoroszlai** [comment]
Thanks @ChenSammi for the patch, @ayushtkn, @kaijchen for the review.

**Pull request merged**